.github/actions/archive-artifacts/action.yaml

```diff
@@ -14,3 +14,29 @@ runs:
         kubectl get events -A -o yaml > /tmp/artifacts/data/${STAGE}/kind-logs/all-events.log;
         kind export logs /tmp/artifacts/data/${STAGE}/kind-logs/kind-export;
         tar zcvf /tmp/artifacts/${{ github.sha }}-${STAGE}-logs-volumes.tgz /tmp/artifacts/data/${STAGE}/kind-logs;
+    - name: Dump kafka
+      shell: bash
+      run: |-
+        set -exu;
+
+        NAMESPACE=${NAMESPACE:-default}
+        KAFKA=$(kubectl get pods -n ${NAMESPACE} -lkafka_cr=${ZENKO_NAME:-end2end}-base-queue -o jsonpath='{.items[0].metadata.name}');
+
+        kubectl exec -itn ${NAMESPACE} ${KAFKA} -c kafka -- \
+            env KAFKA_OPTS= kafka-topics.sh --bootstrap-server :9092 --list \
+            > /tmp/artifacts/data/${STAGE}/kafka-topics.log;
+        kubectl exec -itn ${NAMESPACE} ${KAFKA} -c kafka -- \
+            env KAFKA_OPTS= kafka-consumer-groups.sh --bootstrap-server :9092 --list \
+            | grep -v '^__' \
+            > /tmp/artifacts/data/${STAGE}/kafka-consumer-groups.log;
+        for topic in $(cat /tmp/artifacts/data/${STAGE}/kafka-topics.log); do
+            kubectl exec -itn ${NAMESPACE} ${KAFKA} -c kafka -- \
+                env KAFKA_OPTS= kafka-console-consumer.sh --bootstrap-server :9092 --topic $topic \
+                                                          --from-beginning --timeout-ms 1000 \
+                > /tmp/artifacts/data/${STAGE}/kafka-messages-$topic.log;
+        done;
+        for cg in $(cat /tmp/artifacts/data/${STAGE}/kafka-consumer-groups.log); do
+            kubectl exec -itn ${NAMESPACE} ${KAFKA} -c kafka  -- \
+                env KAFKA_OPTS= kafka-consumer-groups.sh --bootstrap-server :9092 --describe --group $cg \
+                > /tmp/artifacts/data/${STAGE}/kafka-offsets-$cg.log;
+        done;
```

Comment:
Shouldn't this be better in a seperate script file ? Would be easier to maintain and to read..

Reply:
This is a stop-gap measure only. Eventually this (or whatever else we can/need to get from kafka) should be capitalized upon in a real script : this one is not to be reused, and moving to a separate script would lead to extending it, little by little.... until we have created an adhoc solution without thinking of what is needed.

Like we may want to put this in an SosReport plugin ; and/or maybe need to smarter, since we also now have kafka-cleaner which removes messages (which we may need, to troubleshoot "missed" first messages)

So keeping it here for now, to avoid introducing any bias; and we should re-assess if/when we we do followup updates.